src/pci_slot.c

```diff
@@ -66,3 +62,99 @@ void pci_slot_fini(struct pci_slot *slot)
 		free(slot);
 	}
 }
+
+/**
+ * @brief Finds PCI slot by number of the slot.
+ *
+ * @param[in]       slot_number       Number of the slot
+ *
+ * @return Struct with pci slot if successful, otherwise the function returns NULL pointer.
+ */
+static struct pci_slot *find_pci_slot_by_number(char *slot_number)
+{
+	struct pci_slot *slot = NULL;
+	char *temp;
+
+	if (slot_number == NULL)
+		return NULL;
+
+	list_for_each(sysfs_get_pci_slots(), slot) {
+		temp = basename(slot->sysfs_path);
+		if (temp && strncmp(temp, slot_number, PATH_MAX) == 0)
+			return slot;
+	}
+	return NULL;
+}
+
+/**
+ * @brief Sets the slot response.
+ *
+ * @param[in]        slot       Struct with PCI slot parameters.
+ *
+ * @return STATUS_SUCCESS if successful, otherwise a valid status_t status code.
+ */
+static status_t set_slot_parameters(struct pci_slot *slot, struct slot_response *slot_res)
+{
+	struct block_device *bl_device;
+	status_t status = STATUS_SUCCESS;
+	int attention = get_int(slot->sysfs_path, -1, "attention");
+
+	if (attention == -1)
+		return STATUS_INVALID_STATE;
+
+	slot_res->state = attention_to_ibpi(attention);
+	char* slot_num = basename(slot->sysfs_path);
+
+	snprintf(slot_res->slot, PATH_MAX, "%s", slot_num);
+	bl_device = get_block_device_from_sysfs_path(slot->address);
+	if (bl_device)
+		snprintf(slot_res->device, PATH_MAX, "/dev/%s", basename(bl_device->sysfs_path));
+	else
+		snprintf(slot_res->device, PATH_MAX, "(empty)");
+
+	return status;
+}
+
+status_t pci_get_slot(char *device, char *slot_path, struct slot_response *slot_res)
+{
+	struct pci_slot *slot = NULL;
+	struct block_device *block_device = NULL;
+
+	if (device && device[0] != '\0') {
```

Comment:
I would like to have the simplest possible flow, so we need to use device or slot, never both.